tw-tasks-core/src/main/java/com/transferwise/tasks/processing/TasksProcessingService.java

```diff
@@ -113,6 +113,8 @@
   private Instant shutdownStartTime;
   private final Set<Thread> tasksProcessingThreads = new HashSet<>();
   private final Lock tasksProcessingThreadsLock = new ReentrantLock();
+  private final Lock lifecycleLock = new ReentrantLock();
+  private volatile boolean processingStarted;
```

Comment:
Since it's modified and read within lifecycleLock `volatile` is redundant, unless we have other places for it? =) I like the approach recommended by Brian Goets, which is to use `@GuardedBy` or similar for shared state